File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        ans=[]
        dict ={}
        for i in range(len(nums)):
            dict[nums[i]]=i

        i=0
        for i in range(len(nums)):
         if (target-nums[i] in dict and dict [target-nums[i]]!= i):
            ans.append(i)
            ans.append(dict[target-nums[i]])
            break
             
        return ans
        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let ans = [];
    let freq = new Map();
    for (n = 0; n < nums.length; n++) {
        freq.set(nums[n], n);
    }
    for (n = 0; n < nums.length; n++) {
        if (freq.has(target - nums[n]) && freq.get(target - nums[n])!=n) {
            ans.push(n);
            ans.push(freq.get(target - nums[n]));
             return ans;
        }
    }
   
};